### PR TITLE
Allow box meshes to be empty

### PR DESCRIPTION
**Pull request by majosm**
Needed for inducer/grudge#256.

### CODE REVIEW

File: meshmode/mesh/generation.py [hunk 1]
```diff
@@ -1018,8 +1018,8 @@ def generate_box_mesh(axis_coords, order=1, *, coord_dtype=np.float64,
         boundary_tag_to_face = {}
 
     for iaxis, axc in enumerate(axis_coords):
```
Comment (inducer) [inline meshmode/mesh/generation.py:1020]: Add to the docs some text to indicate that zero length is supported here and what that means/how that works.

**Review by inducer — COMMENTED** [review]
Thanks for working on this! This is ready to land from my end once these wrinkles are addressed; I think they should be straightforward.

**Comment by inducer** [comment]
Unsubscribing... @-mention or request review once it's ready for a look or needs attention.

File: meshmode/mesh/generation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (majosm) [reply meshmode/mesh/generation.py:1022]: How's https://github.com/inducer/meshmode/pull/330/commits/c1a8d72b411c81c38072de744d08113015178e32 and https://github.com/inducer/meshmode/pull/330/commits/4b27692ac82d786c2d6e1bcfe4f547ae663a8a2d?

Comment (majosm) [reply meshmode/mesh/generation.py:1022]: https://github.com/inducer/meshmode/pull/330/commits/f73e49aa84ad7068df23c23b231d58177543e0f2

**Comment by inducer** [comment]
LGTM, thanks!

**Pull request merged**